src/Sylius/Bundle/CoreBundle/spec/EventListener/ProductDeletionListenerSpec.php

```diff
@@ -0,0 +1,86 @@
+<?php
+
+/*
+ * This file is part of the Sylius package.
+ *
+ * (c) Paweł Jędrzejewski
+ *
+ * For the full copyright and license information, please view the LICENSE
+ * file that was distributed with this source code.
+ */
+
+declare(strict_types=1);
+
+namespace spec\Sylius\Bundle\CoreBundle\EventListener;
+
+use PhpSpec\ObjectBehavior;
+use Sylius\Component\Core\Model\ProductInterface;
+use Sylius\Component\Core\Promotion\Updater\Rule\ProductAwareRuleUpdaterInterface;
+use Symfony\Component\EventDispatcher\GenericEvent;
+use Symfony\Component\HttpFoundation\RequestStack;
+use Symfony\Component\HttpFoundation\Session\Flash\FlashBagInterface;
+use Symfony\Component\HttpFoundation\Session\SessionInterface;
+
+final class ProductDeletionListenerSpec extends ObjectBehavior
+{
+    function let(
+        RequestStack $requestStack,
+        ProductAwareRuleUpdaterInterface $firstUpdater,
+        ProductAwareRuleUpdaterInterface $secondUpdater,
+    ): void {
+        $this->beConstructedWith($requestStack, $firstUpdater, $secondUpdater);
+    }
+
+    function it_throws_an_exception_when_subject_is_not_a_product(GenericEvent $event): void
+    {
+        $event->getSubject()->willReturn('subject');
+
+        $this->shouldThrow(\InvalidArgumentException::class)->during('removeProductFromPromotionRules', [$event]);
+    }
+
+    function it_does_nothing_when_rule_updaters_do_not_return_any_codes(
+        RequestStack $requestStack,
+        SessionInterface $session,
+        ProductAwareRuleUpdaterInterface $firstUpdater,
+        ProductAwareRuleUpdaterInterface $secondUpdater,
```

Comment:
I would be for keeping the order from `let` method here:
```suggestion
        RequestStack $requestStack,
        ProductAwareRuleUpdaterInterface $firstUpdater,
        ProductAwareRuleUpdaterInterface $secondUpdater,
        SessionInterface $session,
```
the same below